pkg/clients/postgresql/postgresql.go

```diff
@@ -122,8 +123,9 @@ func (c postgresDB) GetConnectionDetails(username, password string) managed.Conn
 // IsInvalidCatalog returns true if passed a pq error indicating
 // that the database does not exist.
 func IsInvalidCatalog(err error) bool {
-	if pqe, ok := err.(*pq.Error); ok {
-		return pqe.Code == pqInvalidCatalog
+	var pqErr *pq.Error
+	if errors.As(err, &pqErr) {
```

Comment:
This is not required with the downgraded `golangci-lint`, however it felt like a best practice worth keeping.